### PR TITLE
Adding atomic bool

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

**Review by shahzilabid — APPROVED** [review]

File: atomicBool/atomicBool.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"sync/atomic"
+	"unsafe"
+)
+
+// Bool is an atomic bool type
+type Bool int32
+
+const (
+	// False state
+	False int32 = iota
```
Comment (OneOfOne) [inline atomicBool/atomicBool.go:13]: type should be Bool not int32

File: atomicBool/atomicBool.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package atomicBool
+
+import (
+	"sync/atomic"
+	"unsafe"
+)
+
+// Bool is an atomic bool type
+type Bool int32
```
Comment (OneOfOne) [inline atomicBool/atomicBool.go:9]: Should have a `Set(b bool)` function imo, feels weird otherwise.

**Review by OneOfOne — CHANGES_REQUESTED** [review]

File: atomicBool/atomicBool.go (hunk 1 above)
Comment (itsmontoya) [reply atomicBool/atomicBool.go:13]: If it's Bool type, then I need to convert it every time I use CompareAndSwap

File: atomicBool/atomicBool.go (hunk 2 above)
Comment (itsmontoya) [reply atomicBool/atomicBool.go:9]: I was playing around with it a bit, I originally had it as Set(state bool). For some reason b.True()/False felt cleaner to use. 

I'm not completely opposed to Set(state) though. @shahzilabid - can we get your input?

File: atomicBool/atomicBool.go (hunk 1 above)
Comment (itsmontoya) [reply atomicBool/atomicBool.go:13]: Example:
```go
// Current
return atomic.CompareAndSwapInt32(b.getIntPtr(), False, True)

// With changes
return atomic.CompareAndSwapInt32(b.getIntPtr(), int32(False), int32(True))
```

Comment (OneOfOne) [reply atomicBool/atomicBool.go:13]: Alright

File: atomicBool/atomicBool.go (hunk 2 above)
Comment (shahzilabid) [reply atomicBool/atomicBool.go:9]: I agree with Ahmed. bool.True()/False() feels a bit awkward

Comment (itsmontoya) [reply atomicBool/atomicBool.go:9]: Okie dokey then! I'll change it

Comment (itsmontoya) [reply atomicBool/atomicBool.go:9]: Changes made, please check!

**Pull request merged**